cocos/2d/CCNode.h

```diff
@@ -1881,9 +1894,17 @@ class CC_DLL Node : public Ref
     mutable bool _additionalTransformDirty; ///< transform dirty ?
     bool _transformUpdated;         ///< Whether or not the Transform object was updated since the last frame
 
-    int _localZOrder;               ///< Local order (relative to its siblings) used to sort the node
+    union {
+        struct {
+            int z; // The original localZOrder
+            unsigned int a; // Order Of Arrival, for avoid sort problem with unstable_sort algorithm.
+        } detail;
+        long long value; // The value to be used in sort
```

Comment:
i am not sure which platform has the issue, but c++ standard doesn't say the actual size of these types.

> What's about: sizeof(int64_t) == sizeof(int32_t) + sizeof(uint32_t)

Yep, it is better. How about std::int64 and std::int32?